qiskit/transpiler/__init__.py

```diff
@@ -40,714 +41,1132 @@
-   operations. In the present case of IBM Q devices, the native gate set can be found by querying
-   the devices themselves, and looking for the corresponding attribute in their configuration:
+Every quantum circuit run on the target device must be expressed using only these instructions.
+For example, suppose one wants to run a simple phase estimation circuit:
 
-   .. code-block::
+.. plot::
+   :include-source:
 
-      from qiskit.providers.fake_provider import FakeVigoV2
-      backend = FakeVigoV2()
+   import numpy as np
+   from qiskit import QuantumCircuit
+   from qiskit.providers.fake_provider import FakeVigoV2
 
-      print(backend.operation_names)
+   backend = FakeVigoV2()
 
-   .. parsed-literal::
+   qc = QuantumCircuit(2, 1)
 
-      ['id', 'rz', 'sx', 'x', 'cx', 'measure', 'delay']
+   qc.h(0)
+   qc.x(1)
+   qc.cp(np.pi/4, 0, 1)
+   qc.h(0)
+   qc.measure([0], [0])
+   qc.draw(output='mpl')
 
-   Every quantum circuit run on an IBM Q device must be expressed using only these basis gates.
-   For example, suppose one wants to run a simple phase estimation circuit:
+We have :math:`H`, :math:`X`, and controlled-:math:`P` gates, all of which are
+not in our devices basis gate set, and must be expanded.  This expansion is taken
+care of for us in the :func:`qiskit.execute` function. However, we can
+decompose the circuit to show what it would look like in the native gate set of
+the IBM Quantum devices (the :class:`~.FakeVigoV2` backend is a fake backend that
+models the historical IBM Vigo 5 qubit device for test purposes):
 
-   .. plot::
-      :include-source:
+.. plot::
+   :include-source:
 
-      import numpy as np
-      from qiskit import QuantumCircuit
-      qc = QuantumCircuit(2, 1)
+   from qiskit import transpile
+   from qiskit import QuantumCircuit
+   from qiskit.providers.fake_provider import FakeVigoV2
 
-      qc.h(0)
-      qc.x(1)
-      qc.cp(np.pi/4, 0, 1)
-      qc.h(0)
-      qc.measure([0], [0])
-      qc.draw(output='mpl')
+   backend = FakeVigoV2()
 
-   We have :math:`H`, :math:`X`, and controlled-:math:`P` gates, all of which are
-   not in our devices basis gate set, and must be expanded.  This expansion is taken
-   care of for us in the :func:`qiskit.execute` function. However, we can
-   decompose the circuit to show what it would look like in the native gate set of
-   the IBM Quantum devices:
+   qc = QuantumCircuit(2, 1)
 
-   .. plot::
-      :include-source:
+   qc.h(0)
+   qc.x(1)
+   qc.cp(np.pi/4, 0, 1)
+   qc.h(0)
+   qc.measure([0], [0])
 
-      import numpy as np
-      from qiskit import QuantumCircuit
-      from qiskit import transpile
-      from qiskit.providers.fake_provider import FakeVigoV2
+   qc_basis = transpile(qc, backend)
+   qc_basis.draw(output='mpl')
 
-      qc = QuantumCircuit(2, 1)
+A few things to highlight. First, the circuit has gotten longer with respect to the
+initial one.  This can be verified by checking the depth of the circuits:
```

Comment:
```suggestion
A few things to highlight. First, the circuit has gotten longer with respect to the
original.  This can be verified by checking the depth of both circuits:
```